src/jit/emitxarch.cpp

```diff
@@ -214,7 +217,8 @@ bool TakesRexWPrefix(instruction ins, emitAttr attr)
     // cased here.
     //
     // Rex_jmp = jmp with rex prefix always requires rex.w prefix.
-    if (ins == INS_movsx || ins == INS_rex_jmp)
+    // vpermq requires w bit to set to 1
+    if (ins == INS_movsx || ins == INS_rex_jmp || ins == INS_vpermq)
     {
```

Comment:
> The VEX.W prefix, I believe, is generally the AVX encoding of the corresponding REX.W prefix.

I agree. From wikipedia, "The W bit is equivalent to the REX prefix's W bit, and specifies a 64-bit operand; for non-integer instructions, it is a general opcode extension bit." But it's for 64-bit register only.

The function `AddRexWPrefix(instruction ins, code_t code)` https://github.com/dotnet/coreclr/blob/master/src/jit/emitxarch.cpp#L350, takes care of both VEX.W bit and REX.W bit. From this aspect, it's OK to let `TakesRexWPrefix` return `true ` for `vpermq` and use `AddRexWPrefix` to set VEX.W bit.

> We do have encodings that use the VEX.W prefix, and it is set based on the opcode size.

For ymm register, VEX.W bit is just a general opcode extension bit. So I think we need to set the VEX W bit explicitly for `vpermq` because `vpermq `uses ymm or xmm. 

I think we have two options here: 
        (1) reuse `TakesRexWPrefix `and `AddRexWPrefix `by letting `TakesRexWPrefix  `return `true `for `vpermq`.
        (2) define two new functions (e.g. `TakesVexWPrefix `and `AddVexWPrefix`) to handle VEX.W bit explicitly. Meanwhile, clean the code in `AddRexWPrefix ` so that `AddRexWPrefix ` only handles REX.W bit.

Any suggestions?